product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -46,15 +46,12 @@ public ProductService(ProductRepository productRepository, MediaService mediaSer
         this.productImageRepository = productImageRepository;
     }
 
-    public List<ProductListVm> getProducts() {
-        return productRepository.findAll().stream()
-                .map(ProductListVm::fromModel)
-                .toList();
-    }
-
-    public ProductListGetVm getAllProducts(int pageNo, int pageSize) {
+    public ProductListGetVm getProductsByBrandOrName(int pageNo, int pageSize, String productName, String brandName) {
```

Comment:
getProductsWithFilter might be a better name, similar for repository

Reply:
i changed the function name. Thanks for your suggestion !